core/src/worker/mod.rs

```diff
@@ -222,15 +221,14 @@ impl Worker {
     pub fn replace_client(&self, new_client: ConfiguredClient<TemporalServiceClientWithMetrics>) {
         // Unregister worker from current client, register in new client at the end
         let mut worker_key = self.worker_key.lock();
+        let slot_provider = (*worker_key).and_then(|k| self.wf_client.workers().unregister(k));
         if let Some(key) = *worker_key {
             self.wf_client.workers().unregister(key);
         }
```

Comment:
Do you need this? Calling unregister twice may not do harm but...

Reply:
Ah thanks! Gonna have to be a new PR sadly.

Reply:
#727 opened